orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/gce/GoogleServerGroupCreator.groovy

```diff
@@ -49,34 +50,39 @@ class GoogleServerGroupCreator implements ServerGroupCreator, DeploymentDetailsA
       operation.credentials = operation.account
     }
 
-    operation.image = operation.image ?: getImage(stage)
-
-    if (!operation.image) {
-      throw new IllegalStateException("No image could be found in ${stage.context.region}.")
+    if (stage.context.imageSource == "artifact") {
+      operation.imageSource = "ARTIFACT"
+      operation.imageArtifact = getImageArtifact(stage)
+    } else {
+      operation.imageSource = "STRING"
```

Comment:
Is string the right noun? What about "legacy"?

Reply:
I chose 'STRING' because we're effectively giving clouddriver the image to deploy either as (1) a string that it needs to resolve in the current project or (2) an artifact.  See https://github.com/spinnaker/clouddriver/blob/935e4d23c22f7832488ad9404b076baad6501560/clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/description/BaseGoogleInstanceDescription.groovy#L46

It's also not completely clear this is legacy, as we'll go into the `else` block when manually starting a cluster and passing an image from the drop-down.